libs/libc/tls/tls_alloc.c

```diff
@@ -57,34 +54,33 @@
 
 int tls_alloc(void)
 {
-  FAR struct tcb_s *rtcb = this_task();
-  FAR struct task_group_s *group = rtcb->group;
-  irqstate_t flags;
+  FAR struct task_info_s *tinfo = task_get_info();
   int candidate;
   int ret = -EAGAIN;
 
-  DEBUGASSERT(group != NULL);
+  DEBUGASSERT(tinfo != NULL);
 
   /* Search for an unused index.  This is done in a critical section here to
    * avoid concurrent modification of the group TLS index set.
    */
 
-  flags = spin_lock_irqsave(NULL);
+  sem_wait(&tinfo->ta_tlssem);
```

Comment:
Should _SEM_WAIT be used?  It should be if this could ever be called from within the OS.  If not, then it should not be built if __KERNEL__ is defined.